### ISSUE TITLE
Panel onDismiss called when Dialog closed

### ISSUE THREAD

**Issue by butchmarshall** [OfficeDev/office-ui-fabric-react#6476]
Reproduced here: [https://codepen.io/anon/pen/oPKgpx](https://codepen.io/anon/pen/oPKgpx)

### Bug Report 
 
- //unpkg.com/office-ui-fabric-react/dist/office-ui-fabric-react.min.js
- Chrome
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (No)
 
Requested priority: (Blocking) 

#### Describe the issue:
 
When Dialog is closed onDismiss for panel is called.

In the reproduction:

1. Click Open Panel
2. Click Open Confirm
3. Click Save/Confirm on the open Dialog

The Panel will also have its onDismiss function called
 
#### Expected behavior:
 
Panel onDismiss should not be called when Dialog is

**Comment by KevinTCoughlin** [comment]
Thank you for the detailed report @butchmarshall and repro examples!

Confirming with the team, but this seems like a bug. Clicking anywhere including the Dialog body closes the Panel behind the Dialog layer when it isn't rendered within the Panel component.

**Comment by KevinTCoughlin** [comment]
> 
> 
> I took a quick glance but don't see anything obvious. This bug has been around for a long time, before Portals, and at least as far back as v6.20.0.

Thank you for checking the history of this bug @JasonGore! Good to know it predates the Portals change 👍. Should help us in tracking this one down.

**Comment by frevds** [comment]
The problem is that the panel dismisses whenever a click is caught outside it's modal layer. 
I trapped a panel into custom dimensions using a Customizer so it does not show fullscreen and although isLightDismiss=false and isBlocking=true and hasCloseButton=false I can dismiss it just by clicking outside of that trap area. So I guess this is what gets triggered as well, when a dialog box is closed. A mouse down outside the area is enough, the panel dismisses even if the mouse is released within the area then.

**Comment by frevds** [comment]
Easiest Workaround: 

add this handler and the issue will disappear:

`
 <Panel
  onOuterClick={ () => {} }>
`

**Comment by micahgodbolt** [comment]
Yeah, since the dialog is outside of the panel it is considered an outerclick, and therefore closes the panel

![image](https://user-images.githubusercontent.com/1434956/50525469-740ba780-0a90-11e9-989c-e6ab4ba9e3fa.png)

Since dialogs will always be rendered in a layer, you can gate outer clicks via onOuterClick (as @frevds said). i.e. there isn't a simple way to know that this dialog came from the panel and to consider it panel content.

**Comment by frevds** [comment]
The workaround works, but it not a desired solution and it certainly shouldn't be like that by design because it is inconsistent to open a modal panel - which by definition is a panel that has no outer zones to click on that close it, modal means that all other UI interaction is supposedly blocked, Foreground stuff within the panel is considered content. It is also incompatible with component paradigms where the panel owner is a component potentially independent of its content components over whose implementation it has no control nor knowledge. If the content of the panel opens a dialog message, a mere click into the message closes the modal panel. This cannot be anticipated and I would consider it a bug to have a modal panel close on its own without explicit action, since this is what should not happen under any circumstances.

**Comment by makafsal** [comment]
Same issue when using onDismiss(). Because when clicking outside the panel area, it will dismiss.

`<Panel
        isOpen={true}
        type={PanelType.large}
        isLightDismiss={false}
        hasCloseButton={false}
        onRenderNavigationContent={this._onRenderHeaderContent}
        focusTrapZoneProps={{
          forceFocusInsideTrap: false
        }}
      >`
